### ISSUE TITLE
What is the estimated time for SOLL to fully support solidity contracts? Is there timeline for the roadmap?

### ISSUE THREAD

**Issue by gongbell** [second-state/SOLL#22]

**Comment by hydai** [comment]
Unfortunately, there is no timeline or roadmap for the Solidity front end.
We have moved our focus from Solidity front end to Yul front end. So we freeze all of the tickets or issues which are related to the Solidity front end.

Therefore, we are going to finish the Yul front end first. After completing full support of Yul contracts, we may go back to implement the Solidity front end in the future but not for now.

To deploy solidity contracts on the ewasm target, here is a workaround: you could use solc to compile solidity contracts into yul IR, and then use SOLL to compile this yul IR into ewasm bytecode.

Also, if you are interested in the Solidity front end, we are happy to have your contribution or support :-)

**Comment by hydai** [comment]
Hi @gongbell 
I found that your research topic `WANA` is very interesting.

We would like to know if there is an essential feature that is required by your research topic? Because we are discussing the solidity front end roadmap and the new plan/timeline will be announced in December 2020.

**Comment by gongbell** [comment]
> Unfortunately, there is no timeline or roadmap for the Solidity front end.
> We have moved our focus from Solidity front end to Yul front end. So we freeze all of the tickets or issues which are related to the Solidity front end.
> 
> Therefore, we are going to finish the Yul front end first. After completing full support of Yul contracts, we may go back to implement the Solidity front end in the future but not for now.
> 
> To deploy solidity contracts on the ewasm target, here is a workaround: you could use solc to compile solidity contracts into yul IR, and then use SOLL to compile this yul IR into ewasm bytecode.
> 
> Also, if you are interested in the Solidity front end, we are happy to have your contribution or support :-)

hi，@dyai, Thanks for your reply. We will try out the solc(solidity2yul)+SOLL(yul to ewam) path.

**Comment by gongbell** [comment]
> Hi @gongbell
> I found that your research topic `WANA` is very interesting.
> 
> We would like to know if there is an essential feature that is required by your research topic? Because we are discussing the solidity front end roadmap and the new plan/timeline will be announced in December 2020.

Thanks a lot. We will try out the solidity front end with complex contracts and feedback essential features. 
btw, is there a current supported language feature list for solidity front end to compile solidity to yul?

**Comment by hydai** [comment]
> Thanks a lot. We will try out the solidity front end with complex contracts and feedback essential features.
> btw, is there a current supported language feature list for solidity front end to compile solidity to yul?

Yes. We have a list about solidity front end based on v0.6.6. Check this: https://github.com/second-state/SOLL/blob/master/doc/guides/FeatureGuideForSolidity.md

**Comment by gongbell** [comment]
> Hi @gongbell
> I found that your research topic `WANA` is very interesting.
> 
> We would like to know if there is an essential feature that is required by your research topic? Because we are discussing the solidity front end roadmap and the new plan/timeline will be announced in December 2020.

Hi, @hydai 
One feature useful for us is the debug hints of the unimplemented features.
When we use solc --ir contract.sol to compile a contract, it will emit UnimplementedFeatureError. However, there is no indication in which line of the solidity file the error is. Is it possible to hint which line the unimplemented feature exists so we can examine detailed features manually?

**Comment by hydai** [comment]
Hi, @gongbell 
SOLL provides some debugging information about unimplemented features.
We have a plan to restart our solidity front end. You can also use our solidity front end to compile and report the unimplemented features to us. And we may increase their priority.

Back to `solc` issues, unfortunately, the `solc` is a compiler made by Ethereum Foundation, you may need to report the UnimplementedFeatureError to [solc](https://github.com/ethereum/solidity/issues) directly for getting more information.

**Comment by gongbell** [comment]
@hydai Thanks you for your suggestions. We will try it out.

**Comment by hydai** [comment]
https://github.com/second-state/SOLL/blob/master/doc/guides/FeatureGuideForSolidity.md#comparison-table-with-solidity-071

We also update the language features table. It will be easier to check our unimplemented features.
